core/tee/tee_svc_cryp.c

```diff
@@ -3428,15 +3441,22 @@ TEE_Result syscall_asymm_verify(unsigned long state,
 
 	switch (TEE_ALG_GET_MAIN_ALG(cs->algo)) {
 	case TEE_MAIN_ALGO_RSA:
-		hash_algo = TEE_DIGEST_HASH_TO_ALGO(cs->algo);
-		res = tee_hash_get_digest_size(hash_algo, &hash_size);
-		if (res != TEE_SUCCESS)
-			break;
-		if (data_len != hash_size) {
-			res = TEE_ERROR_BAD_PARAMETERS;
-			break;
+#ifdef CFG_CRYPTO_RSASSA_NA1
```

Comment:
This ifdef and the one below clutters the code. It seems it would be harmless to remove them. If you agree please remove them, else we'll need to find some other way to avoid having them here.

Reply:
If I'm not mistaken, `TEE_ALG_RSASSA_PKCS1_V1_5` is included [here](https://github.com/EDVTAZ/optee_os/blob/4f0de9c225dca7a16b15aa78238659b254875108/core/tee/tee_svc_cryp.c#L23) in this file, so to remove the ifdef in question, we would need to remove all the ifdefs that guard the include that I linked.

Reply:
This ifdeffing isn't OK so it will need to be fixed. I'm OK with unconditionally including `tee_api_defines_extensions.h`, another option is to put in a helper function `cs->algo != TEE_ALG_RSASSA_PKCS1_V1_5` that would return true if `CFG_CRYPTO_RSASSA_NA1` isn't defined. However, I'd prefer the less complicated solution.

Reply:
I removed the guards